pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -172,28 +172,12 @@ type FilterChainMatchOptions struct {
 // A set of pre-allocated variables related to protocol sniffing logic for
 // propagating the ALPN to upstreams
 var (
-	// These are sniffed by the HTTP Inspector in the outbound listener
-	// We need to forward these ALPNs to upstream so that the upstream can
-	// properly use a HTTP or TCP listener
-	plaintextHTTPALPNs = []string{"http/1.0", "http/1.1", "h2c"}
-	mtlsHTTPALPNs      = []string{"istio-http/1.0", "istio-http/1.1", "istio-h2"}
-
-	mtlsTCPALPNs        = []string{"istio"}
-	mtlsTCPWithMxcALPNs = []string{"istio-peer-exchange", "istio"}
-
-	// These ALPNs are injected in the client side by the ALPN filter.
-	// "istio" is added for each upstream protocol in order to make it
-	// backward compatible. e.g., 1.4 proxy -> 1.3 proxy.
-	mtlsHTTP10ALPN = []string{"istio-http/1.0", "istio"}
-	mtlsHTTP11ALPN = []string{"istio-http/1.1", "istio"}
-	mtlsHTTP2ALPN  = []string{"istio-h2", "istio"}
-
 	// Double the number of filter chains. Half of filter chains are used as http filter chain and half of them are used as tcp proxy
 	// id in [0, len(allChains)/2) are configured as http filter chain, [(len(allChains)/2, len(allChains)) are configured as tcp proxy
 	// If mTLS permissive is enabled, there are five filter chains. The filter chain match should be
 	//  FCM 1: ALPN [istio-http/1.0, istio-http/1.1, istio-h2] Transport protocol: tls      --> HTTP traffic from sidecar over TLS
 	//  FCM 2: ALPN [http/1.0, http/1.1, h2c] Transport protocol: N/A                       --> HTTP traffic over plain text
-	//  FCM 3: ALPN [istio] Transport protocol: tls                                         --> TCP traffic from sidecar over TLS
+	//  FCM 3: ALPN [istio, istio-peer-exchange] Transport protocol: tls                    --> TCP traffic from sidecar over TLS
```

Comment:
This is for removing istio alpn in 1.7? If "istio" is removed, it will break TLS traffic originate from old proxy(<= 1.5) to the new one.